extensions/wikia/AbPerfTesting/js/FrontendDelay.js

```diff
@@ -0,0 +1,81 @@
+/**
+ * Introduce a delayed onDOMReady event to measure the impact of performance regressions
+ *
+  * @see FrontendDelay.class.php
+ */
+(function(w, log, weppy, $) {
+
+	function updateTransactionName(perf_test) {
+		// update Weppy config when transaction context is updated
+		// matches \Transaction::PARAM_AB_PERFORMANCE_TEST const
+		w.wgTransactionContext['perf_test'] = perf_test;
+
+		weppy.setOptions({
+			context: w.wgTransactionContext
+		});
+
+		// set UA custom dimension
+		// should match the PHP code in /extensions/wikia/AbPerfTesting/classes/Hooks.class.php
+		if (perf_test) {
+			w._gaq.push(
```

Comment:
In backend tests we put setting the dimension in TopScripts while in frontend tests we put it at the bottom of `<body>`. Maybe it would be better to be consistent in this aspect and choose a single place?